pyomo/contrib/appsi/solvers/maingo.py

```diff
@@ -0,0 +1,482 @@
+            raise ValueError(
+                f"Unrecognized domain step: {step} (should be either 0 or 1)"
+            )
+
+        return lb, ub, vtype
+
+    def _add_variables(self, variables: List[_GeneralVarData]):
+        for ndx, var in enumerate(variables):
+            varname = self._symbol_map.getSymbol(var, self._labeler)
+            lb, ub, vtype = self._process_domain_and_bounds(var)
+            self._maingo_vars.append(
+                MaingoVar(name=varname, type=vtype, lb=lb, ub=ub, init=var.value)
+            )
+            self._pyomo_var_to_solver_var_id_map[id(var)] = len(self._maingo_vars) - 1
+
+    def _add_params(self, params: List[_ParamData]):
+        pass
+
+    def _reinit(self):
+        saved_config = self.config
+        saved_options = self.maingo_options
+        saved_update_config = self.update_config
+        self.__init__(only_child_vars=self._only_child_vars)
+        self.config = saved_config
+        self.maingo_options = saved_options
+        self.update_config = saved_update_config
+
+    def set_instance(self, model):
+        if self._last_results_object is not None:
+            self._last_results_object.solution_loader.invalidate()
+        if not self.available():
+            c = self.__class__
+            raise PyomoException(
+                f"Solver {c.__module__}.{c.__qualname__} is not available "
+                f"({self.available()})."
+            )
+        self._reinit()
+        self._model = model
+        if self.use_extensions and cmodel_available:
+            self._expr_types = cmodel.PyomoExprTypes()
+
+        if self.config.symbolic_solver_labels:
+            self._labeler = TextLabeler()
+        else:
+            self._labeler = NumericLabeler("x")
+
+        self.add_block(model)
+
+        self._solver_model = maingo_solvermodel.SolverModel(
+            var_list=self._maingo_vars,
+            con_list=self._cons,
+            objective=self._objective,
+            idmap=self._pyomo_var_to_solver_var_id_map,
+        )
+
+    def _add_constraints(self, cons: List[_GeneralConstraintData]):
+        self._cons = cons
+
+    def _add_sos_constraints(self, cons: List[_SOSConstraintData]):
+        if len(cons) >= 1:
+            raise NotImplementedError(
+                "MAiNGO does not currently support SOS constraints."
+            )
+        pass
+
+    def _remove_constraints(self, cons: List[_GeneralConstraintData]):
+        pass
+
+    def _remove_sos_constraints(self, cons: List[_SOSConstraintData]):
+        if len(cons) >= 1:
+            raise NotImplementedError(
+                "MAiNGO does not currently support SOS constraints."
+            )
+        pass
+
+    def _remove_variables(self, variables: List[_GeneralVarData]):
+        pass
+
+    def _remove_params(self, params: List[_ParamData]):
+        pass
+
+    def _update_variables(self, variables: List[_GeneralVarData]):
+        pass
+
+    def update_params(self):
+        pass
```

Comment:
Why pass all of these?

Reply:
We added the missing functionality and no longer pass these functions.